llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -446,6 +446,8 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
     /// inline cost heuristic, but it's a generic cost model to be used in other
     /// places (e.g., in loop unrolling).
     unsigned N = SI.getNumCases();
+    bool HasDefault =
+        !isa<UnreachableInst>(SI.getDefaultDest()->getFirstNonPHIOrDbg());
```

Comment:
Can you move this default-is-unreachable into a helper on SwitchInst instead of repeating it twice?

Reply:
> Can you move this default-is-unreachable into a helper on SwitchInst instead of repeating it twice?

Thanks. I was thinking of doing that, but I wanted to submit a separate PR.

Reply:
I don't think it's major enough to bother splitting out, I'd just do it directly here

Reply:
I've added `defaultDestIsUnreachable`. `SimplifyCFG` and other places of use could be changed to this method in the future.